[ 목표 상세 페이지 / 대시보드 ] 기능 및 디자인 제작

MinaRoh opened the pull request:
## ✅ 작업 내용
목표 상세 페이지를 제작했습니다. /goals/[goalId]로 접근할 수 있습니다.

- 좌측 nav에서 목표 클릭시 해당 목표 상세 페이지로 이동 기능 추가
- 상단 목표 이름, 진행률 표시
- 노트 모아보기 표시
- 노트 모아보기 클릭시 이동 기능 추가
- todo, done 리스트 표시
- todo, done 무한스크롤
- css, 반응형 작업

todo, done list를 가져오는 부분이 대시보드와 겹쳐 대시보드 작업도 진행했습니다.

- 대시보드에 목표 카드 클릭시 해당 목표 상세 페이지로 이동 기능 추가
  - 카드 대신 목표 제목 클릭으로 변경
-  대시보드 더보기 기능 추가


## 📸 스크린샷 / GIF / Link
![image](https://github.com/user-attachments/assets/de88e278-7dd8-4816-a921-8a7de09b35b3)
![image](https://github.com/user-attachments/assets/e2b9aa8d-0142-4598-8915-034cdd9a3392)
![Kapture 2024-10-29 at 16 07 49](https://github.com/user-attachments/assets/9abbd42d-9c77-48eb-8125-e782367e872b)

MinaRoh committed:
🐛 #93 fix: DropdownMenu z-index 추가

- z-10으로 드롭다운 메뉴가 다른 컨텐츠에 가리지 않도록 변경했습니다.

MinaRoh committed:
✨ #93 feat: useGoalQuery 추가

- useInfiniteGoalsQuery에서 여러개의 goals를 가져오기 때문에 불필요한 useGoalsQuery는 삭제했습니다.
- goal id로 한 건의 goal 정보만 가져오는 useGoalQuery를 추가했습니다.

MinaRoh committed:
✨ #93 feat: 목표 상세 페이지 추가

- 목표 상세 이름과 진행도를 받아옵니다.
- 노트 모아보기 영역을 클릭하면 note/[goalId]로 이동합니다.

MinaRoh committed:
🩹 #93 chore: 파일 이름 변경

- 목표 상세 페이지에 목표이름, 진행도를 보여주는 컴포넌트 이름을 GoalTitleWithProgress로 변경했습니다.

MinaRoh committed:
✨ #93 feat: TodoDoneItems 컴포넌트 생성

- 목표 상세 페이지와 대시보드에서 중복으로 사용되는 todo, done list를 TodoDoneItems로 컴포넌트화 했습니다.

MinaRoh committed:
🐛 #93 fix: ProgressBar 컴포넌트 className 추가 가능하도록 수정

- progressbar에 className을 추가할 수 있도록 수정해 대시보드와 목표 상세 페이지에서 커스텀하여 사용할 수 있도록 변경했습니다.

MinaRoh committed:
💄 #93 style: 목표 상세 페이지 반응형 디자인 추가

MinaRoh committed:
✨ #93 feat: 대시보드 목표 클릭시 목표 상세 페이지 이동 기능 추가

MinaRoh committed:
💄 #93 style: 대시보드 더보기 스타일 변경

- 대시보드의 goalTodoCard의 min-h 속성으로 더보기 버튼이 밖으로 나오는 오류를 수정했습니다.

ahrrrl reviewed (approved):
이런... useTodosInfiniteQuery에 { done: false } 인자 넣고 todoitem 해당 data 넣으시면 [TodoDoneItems.tsx](https://github.com/FESI-4-4/slid-todo/pull/96/files#diff-55311d2ff951ba26106cd3e32125295793fb50181a0af9969d8834196bb5a094) 만드실 필요 없으셨을텐데

MinaRoh commented:
> 이런... useTodosInfiniteQuery에 { done: false } 인자 넣고 todoitem 해당 data 넣으시면 [TodoDoneItems.tsx](https://github.com/FESI-4-4/slid-todo/pull/96/files#diff-55311d2ff951ba26106cd3e32125295793fb50181a0af9969d8834196bb5a094) 만드실 필요 없으셨을텐데

화면 영역이 좁아서 무한스크롤을 어떻게 해야하나, 대시보드 더보기 기능과 비슷하게 해야하나 고민했는데 생각 정리돼서 이 pr 닫지 않고 무한스크롤 기능까지 해서 추가 작성하고 merge하겠습니다 😀😀

MinaRoh committed:
✨ #93 feat: 목표 상세 할 일 무한스크롤 기능 추가

- 목표 상세 페이지의 todo, done의 무한스크롤 기능을 추가했습니다.
- todo, done 영역을 각각 표시했던 부분을 배열에 넣고 반복해 중복되는 코드를 줄였습니다.

MinaRoh committed:
🐛 #93 fix: 목표 카드에서 할 일 클릭되지 않고 페이지 이동하는 오류 수정

- 할 일 체크를 클릭해도 목표 카드를 클릭한 것으로 인식되는 등, 매우 작은 부분들까지 목표 상세 페이지로 이동하게 되어 더보기 이외에 영역 클릭시 페이지 이동 조건을 목표 제목 클릭시 이동으로 변경했습니다.

MinaRoh committed:
♻️ #93 refactor: 대시보드, 목표 상세의 할 일 컴포넌트 구분 및 대시보드 더보기 기능 추가

- 대시보드 더보기 버튼의 기능을 추가했습니다. 버튼 클릭시마다 5개의 할 일을 추가로 불러옵니다.
- 대시보드 더보기 / 목표 상세 무한스크롤 등 컴포넌트가 다른 부분이 많아 공용으로 사용하던 TodoDoneItems를 구분하여 각각 파일로 작성했습니다.

MinaRoh committed:
🔀 merge: feature/93 <= develop

MinaRoh committed:
🩹 #93 fix: import 파일 이름 수정
